docs/source/upgrade.rst

```diff
@@ -0,0 +1,747 @@
+.. currentmodule:: chainer
+
+Upgrade Guide from v1 to v2
+===========================
+
+This document provides detailed information for fixing your codes originally written for Chainer v1 to fit much better to Chainer v2.
+It also explains which part of your codes needs fixes when you upgrade Chainer from v1 to v2.
+
+.. contents::
+   :local:
+
+CuPy
+----
+
+.. _upgrade-cupy-separation:
+
+CuPy has been separated from Chainer into a separate package
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CuPy, which was originally a part of Chainer, has been separated into a different Python package since Chainer v2.
+It changes the way to set up Chainer with CUDA support.
+In particular, you have to separately install :mod:`cupy` package to enable CUDA support.
+See :ref:`install-guide` for the recommended installation steps.
+
+Fortunately, there is no need of updating your source code to catch up with this change.
```

Comment:
I think we should make this sentence bold because what users need not to change is as important as what users must change.